core/src/model.jl

```diff
@@ -28,6 +28,78 @@ struct Model{T}
     end
 end
 
+"""
+Get the Jacobian evaluation function via DifferentiationInterface.jl.
+The time derivative is also supplied in case a Rosenbrock method is used.
+"""
+function get_diff_eval(du::Vector, u::Vector, p::Parameters, solver::Solver)
+    (; p_non_diff, diff_cache, p_mutable) = p
+    backend = get_ad_type(solver)
+    sparsity_detector = TracerSparsityDetector()
+
+    backend_jac = if solver.sparse
+        AutoSparse(backend; sparsity_detector, coloring_algorithm = GreedyColoringAlgorithm())
+    else
+        backend
+    end
+
+    t = 0.0
+
+    # Activate all nodes to catch all possible state dependencies
+    p_mutable.all_nodes_active = true
+    jac_prep = prepare_jacobian(
```

Comment:
If you want to be extra-safe, [this huge-ass PR](https://github.com/JuliaDiff/DifferentiationInterface.jl/pull/745) contains additional checking mechanisms, to make sure that your preparation is consistent with your execution. All you have to do to activate them is go `prepare_jacobian(blablabla; strict=Val(true))`. It will be the default in the next breaking release of DI.